File: self-spark-demo/src/main/scala/self/spark/scala/demo/SampleApp.scala
package self.spark.scala.demo

import org.apache.spark.sql.SparkSession

/**
  * SampleApp
  *
  * @author chenzb
  * @date 2020/8/11
  */
class SampleApp {

  def main(args: Array[String]): Unit = {
    val logFile = "YOUR_SPARK_HOME/README.md" // Should be some file on your system
    val spark = SparkSession.builder()
        .master("")
        .appName("Simple Application")
        .config("spark.executor.instances", "10")
        .config("spark.executor.memory", "10g")
        .getOrCreate()
    val logData = spark.read.textFile(logFile).cache()
    val numAs = logData.filter(line => line.contains("a")).count()
    val numBs = logData.filter(line => line.contains("b")).count()
    println(s"Lines with a: $numAs, Lines with b: $numBs")
    spark.stop()
  }


}

File: self-spark-demo/src/main/java/self/spark/java/demo/SampleApp.java
package self.spark.java.demo;

import org.apache.spark.api.java.function.FilterFunction;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;

/**
 * SampleApp
 *
 * @author chenzb
 * @date 2020/8/11
 */
public class SampleApp {

	public static void main(String[] args) {
//		String logFile = "/opt/uyun/databank/spark/spark/README.md"; // Should be some file on your system
		String logFile = "C:\\Users\\chenzb\\Desktop\\文档\\维度管理\\流程.txt"; // Should be some file on your system
		SparkSession spark = SparkSession.builder().appName("Simple Application").getOrCreate();
		Dataset<String> logData = spark.read().textFile(logFile).cache();

		long numAs = logData.filter((FilterFunction<String>) s -> s.contains("a")).count();
		long numBs = logData.filter((FilterFunction<String>) s -> s.contains("b")).count();

		System.out.println("Lines with a: " + numAs + ", lines with b: " + numBs);
		spark.stop();
	}
}
